# This config file contains the configuration values that are used by the Panda task,
# robot and Gazebo goal environments. Variables marked with the (DEFAULT) tag are
# overloaded by PandaTaskEnv constructor arguments.

#########################################
# Action space parameters ##############
#########################################
action_space:
  use_gripper_width: True # (DEFAULT) - Whether to use gripper_width instead of individual finger joint commands
  bounds:
    ee_pose:
      low:
        x: -1.3
        y: -1.3
        z: 0.0
        rx: 0
        ry: 0
        rz: 0
        rw: 0
      high:
        x: 1.3
        y: 1.3
        z: 1.3
        rx: 1
        ry: 1
        rz: 1
        rw: 1
    joint_positions: # NOTE: The position and effort limits can be found in the robot urdf file (see franka_description).
      low:
        panda_joint1: -2.8973
        panda_joint2: -1.7628
        panda_joint3: -2.8973
        panda_joint4: -3.0718
        panda_joint5: -2.8973
        panda_joint6: -0.0175
        panda_joint7: -2.8973
        panda_finger_joint1: 0.0
        panda_finger_joint2: 0.0
        gripper_width: 0.0
      high:
        panda_joint1: 2.8973
        panda_joint2: 1.7628
        panda_joint3: 2.8973
        panda_joint4: -0.0698
        panda_joint5: 2.8973
        panda_joint6: 3.7525
        panda_joint7: 2.8973
        panda_finger_joint1: 0.04
        panda_finger_joint2: 0.04
        gripper_width: 0.2
    joint_efforts:
      low:
        panda_joint1: -87
        panda_joint2: -87
        panda_joint3: -87
        panda_joint4: -87
        panda_joint5: -12
        panda_joint6: -12
        panda_joint7: -12
        panda_finger_joint1: -20
        panda_finger_joint2: -20
      high:
        panda_joint1: 87
        panda_joint2: 87
        panda_joint3: 87
        panda_joint4: 87
        panda_joint5: 12
        panda_joint6: 12
        panda_joint7: 12
        panda_finger_joint1: 20
        panda_finger_joint2: 20

#########################################
# Training related parameters ###########
#########################################
training:
  # Main training parameters
  reward_type: "Sparse" # (DEFAULT) - The reward type used in the reward function ("sparse" or "dense").
  distance_threshold: 0.05 # The threshold for determining a goal has been reached
  has_object: False # (DEFAULT) - Train with object

  # Settings for setting up the object
  object_sampling:
    distance_threshold: 0.1 # Threshold for determining whether new object pose is far enough from old obj pose
    init_obj_pose: # (DEFAULT) -  The initial pose of the grasp object.
      x: 0.4
      y: 0.0
      z: 0.0
      rx: 0.0
      ry: 0.0
      rz: 0.0
      rw: 1.0
    visualize_bounds: True # Turn object sampling region visualization on and off
    bounds: # (DEFAULT) - The bounds that define the region, rel. to the 'init_obj_pose' in which the object pose is sampled.
      x_min: -0.7
      x_max: 0.7
      y_min: -0.7
      y_max: 0.7

  # Settings for sampling of the target (Goal position)
  target_sampling:
    # Main settings
    strategy: "global" # Global or local goal target sampling
    target_in_the_air: False # (DEFAULT) - Whether the object place target can be in the air
    visualize_target: True # Turn on and off rviz target visualization
    visualize_bounds: True # Turn target sampling region visualization on and off
    target_offset: # (DEFAULT) - An additional offset added to the target
      x: 0.0
      y: 0.0
      z: 0.0

    # Target sampling bounds
    bounds: # (DEFAULT) - The bounds that define the region in which the target pose is sampled
      global: # The global sample bounds relative to the world frame
        x_min: -0.7
        x_max: 0.7
        y_min: -0.7
        y_max: 0.7
        z_min: 0.0
        z_max: 1.3
      local: # The target sample bounds relative to the current ee_pose.
        x_min: -0.15
        x_max: 0.15
        y_min: -0.15
        y_max: 0.15
        z_min: -0.15
        z_max: 0.15

#########################################
# Simulation parameters #################
#########################################
simulation:
  # Main simulation settings
  reset_robot_pose: True # Whether we want to reset the pose when the environment is reset

  # Control settings
  control:
    robot_arm_control_type: "position_control" # (DEFAULT) - The type of control
    robot_hand_control_type: "position_control" # (DEFAULT) - The type of control
    block_gripper: False # (DEFAULT) - Block the gripper control
    ee_link: "panda_hand" # Link that is specified as the end effector. Used for the distance to the target position.
    gripper_extra_height: 0.10 # (DEFAULT) - Add extra height

  # Initial pose given in generalized coordinates (q)
  init_pose_sampling:
    random_init_pose: True # Use random pose when resetting the simulation
    randomize_first_episode: True #  Also randomize the pose in the first episode
    visualize_bounds: True # Turn init pose sampling region visualization on and off
    init_robot_pose: # (DEFAULT) - The initial robot pose
      x: 0.23
      y: 0.29
      z: 0.35
      rx: 0.78
      ry: 0.62
      rz: -0.0
      rw: 4.42
      gripper_width: 0.0
